DTML.EduBot/Common/TraceSourceExceptionLogger.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Web;
+using System.Web.Http.ExceptionHandling;
+
+namespace DTML.EduBot.Common
+{
+    public class TraceSourceExceptionLogger : ExceptionLogger
+    {
+        private readonly TraceSource _traceSource;
+
+        public TraceSourceExceptionLogger(TraceSource traceSource)
+        {
+            _traceSource = traceSource;
+        }
+
+        public override void Log(ExceptionLoggerContext context)
+        {
+            _traceSource.TraceEvent(TraceEventType.Error, 1,
+                "Unhandled exception processing {0} for {1}: {2}, w/ user ID: {3}",
+                context.Request.Method,
+                context.Request.RequestUri,
+                context.Exception,
+                context.Exception.Data["id"]);
+
+            Console.WriteLine(context.Exception.Data["id"]);
```

Comment:
We should not log exception message into the console, but rather some telemetry. Look at Deepak/AppInsight branch https://github.com/eklavyamirani/DTML.EduBot/tree/Deepak/AppInsight for more information,